core/src/test/scala/quasar/destination/s3/S3DestinationSpec.scala

```diff
@@ -100,7 +100,9 @@ final class MockUpload(status: Ref[IO, Map[ObjectKey, String]]) extends Upload[I
   def upload(bytes: Stream[IO, Byte], bucket: Bucket, key: ObjectKey): Stream[IO, Unit] =
     for {
       data <- bytes.through(text.utf8Decode)
-      _ <- Stream.eval(status.update(_ + (key -> data)))
+      _ <- Stream.eval(status.update(currentStatus =>
+        currentStatus.get(key).fold(currentStatus + (key -> data))(existingVal =>
+          currentStatus + (key -> (existingVal + data)))))
```

Comment:
⛳️ This is actually equivalent to what you had before, except tossing `foldMonoid` on the result of `through`.

Reply:
How? I'm not sure where the monoid comes from.

The previous implementation overwrote the status with each new chunk. This implementation concatenates it. Anyway, I pushed an implementation using `foldMonoid`.